frontend/src/Calendar.js

```diff
@@ -2,19 +2,108 @@ import React from 'react'
 import FullCalendar from '@fullcalendar/react'
 import dayGridPlugin from '@fullcalendar/daygrid'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
+// the conversion of one minute to 60000 milliseconds
+const MIN_TO_MS = 60000;
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
+      schedulesTimes: [],
+      scheduleIdx: 0     // Current schedule
```

Comment:
nit: `selectedScheduleId` or `selectedScheduleIndex` or similar

-> id or index over idx
-> being more specific about what this schedule index really means

Reply:
Thanks for the suggestion. I went with `selectedScheduleId`